### PR TITLE
feat: add login UI

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:**
- Made the form for user login, with input of email, password, google authorization button and continue button.
- Made it possible to display and hide the password.
- Added field validation.
- Made a google authorization button.
- Made it possible to click the "Continue" button only when all fields are filled.

[Link to Jira ticket](https://smartdc23.atlassian.net/browse/SMAR-36?atlOrigin=eyJpIjoiYmI2NTBiNDI0NWJkNDU1YWE5MDkxYTA3Mjk4ZmFjOTAiLCJwIjoiaiJ9)

**PR checklist:**
1.1 Common
1.1.1 types for input and output params
1.1.2 no `any`, should be also added to eslint rules
1.1.3 try/catch for any async function
1.1.4 no `magic` numbers
1.1.5 compare only with constants not with strings
`(instead of user === ‘admin’, better user === roles.admin)`
1.1.6 no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
1.1.7 avoid similar types with duplicating by generics
1.1.8 no `console.log` in pr
1.1.9 .env file should be in .gitignore
1.1.10 delete commented code if it's not part of documentation
1.1.11 no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
1.1.12 `constants` and `function` names should be lowercase.
1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
1.1.14 import rules. imports should come in a specific order: node modules, absolute
path, relative path
1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.3 Frontend|Mobile
1.3.1 split component and logic (move logic to custom hook file)
1.3.2 colors, font size, and font name should be in the theme or in the constants
1.3.3 no text in the components, use i18n approach, even if you have only one language for now
1.3.4 inline styles prohibited
1.3.5 import should be absolute. instead of `../../../components/myComponent` should be
`components/myComponent`

**General view:**
image
**Error in fields:**
image
**Error in field:**
image
**See the password:**
image
**The ability to click the "Continue" button only when all fields are filled in correctly:**
image

### CODE REVIEW

File: src/components/Auth/LoginForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    resolver: yupResolver(LoginSchema),
+  });
+  useEffect(() => {
+    register('password');
+  }, []);
+
+  const onSubmit = (data: ILogin) => { };
+
+  const login = useGoogleLogin({ });
```
Comment (lyudad) [inline src/components/Auth/LoginForm/index.tsx]: you have mess, states should be with states, hooks with hooks, handlers with handlers

File: src/components/Auth/LoginForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+              type={showPassword ? "text" : "password"}
+              placeholder={t("Auth.enterPassword")??""}
+              helperText={errors.password?.message}
+              error={Boolean(errors?.password)}
+              required={true}
+              InputProps={{
+                endAdornment: (
+                  <IconButton onClick={handleClickShowPassword}>
+                    <InputAdornment position={end}>
+                      {showPassword ? <PasswordImg src={visible}/> : <PasswordImg src={visibleOff}/>}
```
Comment (lyudad) [inline src/components/Auth/LoginForm/index.tsx]: <PasswordImg src={howPassword ?visible : visibleOff }/>

File: src/components/Auth/LoginForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                    </InputAdornment>
+                  </IconButton>
+                ),
+              }}
+            />
+          </AuthInput>
+          <AuthGoogleContainer onClick={() => login()}>
```
Comment (lyudad) [inline src/components/Auth/LoginForm/index.tsx]: <AuthGoogleContainer onClick={login}>

File: src/components/Auth/LoginForm/index.tsx (hunk 2 above)
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/index.tsx]: solved

File: src/components/Auth/LoginForm/index.tsx (hunk 1 above)
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/index.tsx]: put google login element to separate file

File: src/components/Auth/LoginForm/index.tsx (hunk 3 above)
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/index.tsx]: put to separate GoogleLoginButton element

File: src/components/Auth/LoginForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    control,
+    formState: { errors, isValid }
+  } = useForm<ILogin>({
+    mode: 'onChange',
+    defaultValues: {
+      email: '',
+      password: ''
+    },
+    resolver: yupResolver(LoginSchema),
+  });
```
[the remark this answers is not in the document]
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/index.tsx:48]: Your screenshot is still not loaded - what is wrong here?

[the remark this answers is not in the document]
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/index.tsx:48]: add blank line before resolver

File: src/components/Auth/LoginForm/GoogleLogin.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { AuthGoogleContainer, GoogleImg, GoogleText } from "../styles";
+import { useTranslation } from "react-i18next";
+import google from '@assets/auth/google.svg'
+
+function GoogleLoginButton() {
+    const { t } = useTranslation();
+    const login = useGoogleLogin({});
+  
+    return (
+        <AuthGoogleContainer onClick={() => login()}>
```
Comment (lyudad) [inline src/components/Auth/LoginForm/GoogleLogin.tsx]: onClick={login}

File: src/validation/auth.validate.ts [hunk 6]
```diff
@@ -34,3 +34,17 @@ export const signUpSchema = yup.object().shape({
...
+  email: yup
+    .string()
+    .email('*Invalid email format')
+    .required('*Email is required'),
+  password: yup
+    .string()
+    .required('*Please Enter your password')
+    .matches(
+      PASSWORD_PATTERN,
+      '*Must Contain 10 Characters, One Uppercase, One Lowercase',
```
Comment (lyudad) [inline src/validation/auth.validate.ts]: 10 should be pass ass parameter

Comment (SwiftXLight) [reply src/validation/auth.validate.ts]: fixed

File: src/components/Auth/LoginForm/GoogleLogin.tsx (hunk 5 above)
Comment (SwiftXLight) [reply src/components/Auth/LoginForm/GoogleLogin.tsx]: make handler for login

**Review by lyudad — APPROVED** [review]

**Pull request merged**